src/test/regress/expected/multi_mx_function_call_delegation.out

```diff
@@ -322,26 +283,214 @@ PL/pgSQL function mx_call_func(integer,integer) line 8 at assignment
+PL/pgSQL function mx_call_func(integer,integer) line 8 at assignment
+ mx_call_func 
+--------------
+           29
+(1 row)
+
+-- non-const parameter can be pushed down
+select mx_call_func((select x + 1 from mx_call_add(3, 4) x), 2);
+DEBUG:  pushing down the function call
+ERROR:  there is no parameter $0
+CONTEXT:  while executing command on localhost:57638
+-- volatile parameter cannot be pushed down
+select mx_call_func(floor(random())::int, 2);
+DEBUG:  arguments in a distributed function must be constant expressions
+DEBUG:  generating subplan 31_1 for subquery SELECT sum((t1.val OPERATOR(pg_catalog.+) t2.val)) AS sum FROM (multi_mx_function_call_delegation.mx_call_dist_table_1 t1 JOIN multi_mx_function_call_delegation.mx_call_dist_table_2 t2 ON ((t1.id OPERATOR(pg_catalog.=) t2.id)))
+CONTEXT:  SQL statement "SELECT y + (select sum(t1.val + t2.val) from multi_mx_function_call_delegation.mx_call_dist_table_1 t1 join multi_mx_function_call_delegation.mx_call_dist_table_2 t2 on t1.id = t2.id)"
+PL/pgSQL function mx_call_func(integer,integer) line 8 at assignment
+DEBUG:  Plan 31 query after replacing subqueries and CTEs: SELECT (1 OPERATOR(pg_catalog.+) (SELECT intermediate_result.sum FROM read_intermediate_result('31_1'::text, 'binary'::citus_copy_format) intermediate_result(sum bigint)))
+CONTEXT:  SQL statement "SELECT y + (select sum(t1.val + t2.val) from multi_mx_function_call_delegation.mx_call_dist_table_1 t1 join multi_mx_function_call_delegation.mx_call_dist_table_2 t2 on t1.id = t2.id)"
+PL/pgSQL function mx_call_func(integer,integer) line 8 at assignment
+ mx_call_func 
+--------------
+           27
+(1 row)
+
+-- test forms we don't distribute
+select * from mx_call_func(2, 0);
+DEBUG:  generating subplan 33_1 for subquery SELECT sum((t1.val OPERATOR(pg_catalog.+) t2.val)) AS sum FROM (multi_mx_function_call_delegation.mx_call_dist_table_1 t1 JOIN multi_mx_function_call_delegation.mx_call_dist_table_2 t2 ON ((t1.id OPERATOR(pg_catalog.=) t2.id)))
+CONTEXT:  SQL statement "SELECT y + (select sum(t1.val + t2.val) from multi_mx_function_call_delegation.mx_call_dist_table_1 t1 join multi_mx_function_call_delegation.mx_call_dist_table_2 t2 on t1.id = t2.id)"
+PL/pgSQL function mx_call_func(integer,integer) line 8 at assignment
+DEBUG:  Plan 33 query after replacing subqueries and CTEs: SELECT (3 OPERATOR(pg_catalog.+) (SELECT intermediate_result.sum FROM read_intermediate_result('33_1'::text, 'binary'::citus_copy_format) intermediate_result(sum bigint)))
+CONTEXT:  SQL statement "SELECT y + (select sum(t1.val + t2.val) from multi_mx_function_call_delegation.mx_call_dist_table_1 t1 join multi_mx_function_call_delegation.mx_call_dist_table_2 t2 on t1.id = t2.id)"
+PL/pgSQL function mx_call_func(integer,integer) line 8 at assignment
+ y  
+----
+ 29
+(1 row)
+
+select mx_call_func(2, 0) from mx_call_dist_table_1;
+ERROR:  cached metadata for shard 102033 is inconsistent
```

Comment:
Is this wrong?

Reply:
Still looking into this, but this also happens earlier in multi_mx_call too: https://github.com/citusdata/citus/pull/3021/files#diff-0dfffc23c4864f2642ff50f1cb57e3a7R445. It happens in one of the workers. It didn't happen before. Still looking into that.

Reply:
I think this is due to `stop_metadata_sync_to_node()`/`start_metadata_sync_to_node()` calls, which recreate metadata but doesn't invalidate the caches. Still looking.